File: frontend/src/objects/rider/profile/RiderIdentity.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type {
  AvailabilityLabel,
  PersonName,
  RiderId,
  VehicleLabel,
  ZoneLabel,
} from '@/objects/core/SharedObjects'

export type RiderIdentity = {
  id: RiderId
  name: PersonName
  vehicle: VehicleLabel
  zone: ZoneLabel
  availability: AvailabilityLabel
}

File: backend/src/main/scala/services/rider/objects/RiderIdentity.scala
package services.rider.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given

import system.objects.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class RiderIdentity(
    id: RiderId,
    name: PersonName,
    vehicle: VehicleLabel,
    zone: ZoneLabel,
    availability: AvailabilityLabel,
)
object RiderIdentity:
  given Encoder[RiderIdentity] = deriveEncoder
  given Decoder[RiderIdentity] = deriveDecoder
